holoviews/plotting/plot.py

```diff
@@ -1071,14 +1199,15 @@ def _update_subplot(self, subplot, spec):
             subplot.overlay_dims = util.OrderedDict(new_dims)
 
 
-    def get_extents(self, overlay, ranges):
-        extents = []
+    def get_extents(self, overlay, ranges, range_type='combined'):
```

Comment:
This method is a bit of a monster...

Reply:
You should see it before I refactored it :-) I'll have another go at factoring stuff out though.